multiaddr/multiaddr.py

```diff
@@ -37,16 +37,12 @@ def __init__(self, string_addr=None, bytes_addr=None):
 
         Only one of string_addr or bytes_addr may be set
         """
-        if string_addr is not None and bytes_addr is not None:
-            raise ValueError(
-                "Only one of 'string_addr' or 'bytes_addr' may be set")
-
-        if string_addr is not None:
+        if string_addr is not None and bytes_addr is None:
             self._bytes = string_to_bytes(string_addr)
-        elif bytes_addr is not None:
+        elif bytes_addr is not None and string_addr is None:
             self._bytes = bytes_addr
         else:
-            raise ValueError("Invalid address type, must be bytes or str")
+            raise ValueError("Invalid address type, must be bytes xor str")
```

Comment:
Understood.  I struggled a bit with that.  Yes, it would be much better if only one parameter was taken as an input, but like you said not sure of a good way to do that works in both python 2 and 3.